src/classes/UTIL_OrgTelemetry_SObject_BATCH.cls

```diff
@@ -214,16 +249,40 @@ public class UTIL_OrgTelemetry_SObject_BATCH implements Database.Batchable<SObje
      * @description When the entire batch job is done, submit the final count variables to the LMO
      */
     public void finish(Database.BatchableContext context) {
-        switch on (jobMode) {
-            when OPPORTUNITIES {
-                featureManager.setPackageIntegerValue(UTIL_OrgTelemetry_SVC.TelemetryParameterName.Data_CountRdOppsOpenEnded.name(), rdOpenStatusOppsCount);
-                featureManager.setPackageIntegerValue(UTIL_OrgTelemetry_SVC.TelemetryParameterName.Data_CountRdOppsAll.name(), rdAllOppsCount);
+        if (shouldChainNextBatch()) {
+
+            switch on (jobMode) {
+                when OPPORTUNITIES {
+                    Database.executeBatch(this, OPPORTUNITIES_BATCH_SIZE);
```

Comment:
I haven't seen this pattern before and can't find it documented anywhere. Does chaining with `this` carry over the current stateful vars without having to pass them all in as parameters into the constructor? Assuming that part works, are there any potential issues with this over creating a new instance of the class and setting the stateful vars?

Reply:
@jbrandolinisf actually showed this to me. I haven't run into any issues over creating a new instance. I believe it does carry over the current stateful vars, I tested this by doing the following:

* Add the assertion `system.assert(false, ''+rdAllCount);`  in the execute method in `UTIL_OrgTelemetry_SObject_BATCH`

* Run the following in a test method (so that async batch executes)

```
 UTIL_OrgTelemetry_SObject_BATCH batch =
            new UTIL_OrgTelemetry_SObject_BATCH(UTIL_OrgTelemetry_SObject_BATCH.TelemetryBatchJobMode.RECURRING_DONATIONS);
        batch.lastProcessedId = recurringDonations[0].Id;
        batch.featureManager = (UTIL_FeatureManagement) Test.createStub(UTIL_FeatureManagement.class, featureManagementMock);

        batch.rdAllCount = 10;

        Test.startTest();
        batch.finish(null);
        Test.stopTest();
```

* Assertion should output `10`